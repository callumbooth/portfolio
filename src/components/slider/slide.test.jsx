import React from "react";
import renderer from "react-test-renderer";
import Slide from "./slide";
import { MemoryRouter } from "react-router-dom";

it("Slide renders correctly", () => {
  const data = {
    name: "Super theme",
    slug: "super-theme",
    rotate: 0,
    featured: true,
    tags: ["tag", "tag2", "tag3"],
    launched: "September 2018",
    summary: "A Moodle theme to end all Moodle themes",
    content: "content goes here",
  };

  const component = renderer.create(
    <MemoryRouter>
      <Slide
        delayTransition={() => {}}
        loaded={true}
        data={data}
        toggleInfo={() => {}}
        currentSlide={1}
      />
    </MemoryRouter>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
