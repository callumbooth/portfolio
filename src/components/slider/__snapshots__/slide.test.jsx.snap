// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide renders correctly 1`] = `
<div
  className="slide fadeOut"
>
  <div
    className="background-wrapper-top"
  >
    <svg
      viewBox="0 0 1520 1080"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <line
          className="a l1"
          transform="translate(1090 0) rotate(0 217.5 217.5)"
          x2={435}
          y2={435}
        />
        <line
          className="a l2"
          transform="translate(417 0) rotate(0 184.5 184.5)"
          x2={369}
          y2={369}
        />
      </g>
    </svg>
  </div>
  <div
    className="slide-wrapper"
  >
    <div
      className="slide-title"
    >
      <a
        href="projects/super-theme"
        onClick={[Function]}
      >
        Super theme
      </a>
    </div>
    <div
      className="moreinfo"
    >
      <button
        className="btn btn-red"
        onClick={[Function]}
      >
        i
      </button>
      <div
        className="info"
      >
        <p>
          A Moodle theme to end all Moodle themes
        </p>
        <a
          className="project-link text-uppercase text-white text-small"
          href="projects/super-theme"
          onClick={[Function]}
        >
          View Project
        </a>
      </div>
    </div>
  </div>
  <div
    className="background-wrapper-bottom"
  >
    <svg
      viewBox="0 0 1520 1080"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          className="b b1"
          d="M0,0 l919,919 h-218 l-701,-701 Z"
          transform="translate(0 190) rotate(0 459.5 459.5)"
        />
        <line
          className="a l3"
          transform="translate(0 410) rotate(0 350.5 350.5)"
          x2={701}
          y2={701}
        />
        <line
          className="a l4"
          transform="translate(0 190) rotate(0 460.5 460.5)"
          x2={921}
          y2={921}
        />
        <line
          className="a l5"
          transform="translate(1086 669) rotate(0 205.5 205.5)"
          x2={411}
          y2={411}
        />
        <line
          className="a l6"
          transform="translate(1008 873)"
          x2={207}
          y2={207}
        />
      </g>
    </svg>
  </div>
</div>
`;
